### ISSUE TITLE
Guest login prompt

### ISSUE THREAD

**Issue by pedro16v** [catalpainternational/canoe#1015]
## Overview

The content on some instances of Bero can be accessed by unauthenticated users. When users open the app, they're informed they are using the app as guests and are invited to login to keep their progress. We need a better looking prompt.

Additionally, certain options under **Profile** and **Settings** are not accessible.

### 1. New login prompt
| Current state | New prompt |
| --- | --- | 
![Screenshot 2021-09-14 at 14 08 44](https://user-images.githubusercontent.com/2813677/133198413-54931025-16c9-4bea-b18e-604a0b1a0b66.png) | ![Screenshot 2021-09-14 at 14 08 02](https://user-images.githubusercontent.com/2813677/133198421-155649cb-a825-4d6e-a765-374494a87787.png)

### 2. Profile and Settings

| Non-logged in Profile | Non-logged in Settings |
| --- | --- | 
![Screenshot 2021-09-14 at 14 15 00](https://user-images.githubusercontent.com/2813677/133199028-6f78a280-1163-43a8-85cd-2d5b2b6a174f.png) | ![Screenshot 2021-09-14 at 14 15 08](https://user-images.githubusercontent.com/2813677/133199026-e9164da4-48b6-4240-a7a7-070dda226b24.png)

## Resources

- Guest login prompt A — [InVision](https://invis.io/V411JI0JRBCZ)
- Guest login prompt B — [InVision](https://invis.io/4W11P8LARVBY)
- Guest Profile - [InVision](https://invis.io/MD113620ZJS6#/456887198_Profile_-_Non_Logged)
- Guest Settings - [InVision](https://invis.io/MD113620ZJS6#/456887199_Settings_-_Non_Logged)

**Comment by Juliana-t** [comment]
@andrewwapling just to make things clear for the implementation, when do we show:
- Guest login prompt A — [InVision](https://invis.io/V411JI0JRBCZ)
- Guest login prompt B — [InVision](https://invis.io/4W11P8LARVBY)
I will assume prompt A to be shown always as an alert banner, but not sure when are we showing B

**Comment by pedro16v** [comment]
I believe these are alternatives 🤔

**Comment by andrewwapling** [comment]
Hi @Juliana-t @pedro16v the different versions above were variants of the same feature. I initially created the top banner pattern however when discussing with David we felt the roll up modal from the bottom created more affordance to the user of the need to login. My understanding is that we'd like to push users to login (if possible) to enjoy the full Bero experience. 

In terms of the when and how, my thinking here is that we could show this to a user on their first visit to the home page when they scroll down. In an effort to not overwhelm a user we could perhaps look to only show this to a non-logged user once per session?

**Comment by Juliana-t** [comment]
@andrewwapling sounds good. Showing to non-logged users once per session. If the user clicks on 'Continue without login' should we keep showing the top banner (with a link to login) in the rest of the session? I am thinking that for a reason the non logged him user may close that roll up and then there is no other button/call to action/way to go to the login page. For that reason, I would recommend your approach, and if the user decides to skip, keeping that top banner. What do you think?

**Comment by andrewwapling** [comment]
@Juliana-t yes I also like this approach, I likewise believe the banner at top would be a good fail safe approach to reinforce the notion of not being logged in to a user. My only other thought is whether we allow the top notification to be user dismissible or keep it persistent. The answer probably lies in how strongly we would like to encourage users to login. What are your thoughts here?

**Comment by Juliana-t** [comment]
I think we should keep it - I believe the desirable action is for users to login, except when they are offline. If they are offline, and if it is not too much of an effort to implement, I would keep that top bar just informative (You are not logged in, your progress won't be synchronised with the server, login when you are online, something like that) and make in that case add an option to dismiss the message.
@PeteCoward what are your thoughts?

**Comment by PeteCoward** [comment]
Bero has always supported offline use, and the saving of actions when offline, either logged in or out.
If a user completes content or exams while offline, it will be sent to the server eventually once they get back online.
If a user does stuff while not logged in, it will be saved on that device, and sent to the server once they are logged in.

A login just allows the saving of actions and completion across devices, or over a logout / login cycle, and communication of actions to the server and administrators.

With regards to showing different messages if online and if offline, be aware that online/offline detection is not very reliable, in the cases where it might fail the attempt to communicate this information can end up causing confusion.

In summary I would keep it simple, and as we already do allow as much offline functionality as possible. Login/password reset/account profile changes are examples of things we can only do online.

**Comment by andrewwapling** [comment]
Hi @PeteCoward @megganeturner @pedro16v the guest login alert (Guest login prompt A — [InVision](https://invis.io/V411JI0JRBCZ)) sits on top of the content as opposed to pushing the content down. Is it possible to change this so that it pushes the content below down and when dismissed rolls it back out of the way. Apart from a more enjoyable interaction it also avoids, awkward cutting off of content below.
